### PR TITLE
UCP/API/UCT/CONFIG/TEST: Multiple cleanups.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/496/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1068/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1070/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/498/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
autobot build

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -161,7 +161,8 @@ typedef struct ucp_generic_dt_ops {
      * @ingroup UCP_DATATYPE
      * @brief Start a packing request.
      *
-     * The pointer refers to application defined start-to-pack routine.
+     * The pointer refers to application defined start-to-pack routine. It will
+     * be called from the send function.
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: This routine is invoked from the @ref ucp_tag_send_nb "send" routine.

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -176,7 +177,8 @@ typedef struct ucp_generic_dt_ops {
      * @ingroup UCP_DATATYPE
      * @brief Start an unpacking request.
      *
-     * The pointer refers to application defined start-to-unpack routine.
+     * The pointer refers to application defined start-to-unpack routine. It will
+     * be called from the receive function.
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: This routine is invoked from the @ref ucp_tag_recv_nb "send" routine.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/500/ for details (Mellanox internal link).

File: src/ucp/api/ucp.h (hunk 2 above)
Comment (shamisp) [inline src/ucp/api/ucp.h]: "receive" routine

**Comment by yosefe** [comment]
yes, i will also send alter to mailing list about config prefix change.

**Comment by miked-mellanox** [comment]
:+1:

**Comment by shamisp** [comment]
Once the wording of the documentation is changed, I'm overall ok with this PR.

**Comment by yosefe** [comment]
@miked-mellanox actually i don't see any mtt settings today which are affected by this change. mtt is only using UCX_ which stays the same.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1076/ for details.

**Comment by shamisp** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/506/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
``` shell
-x UCX_TLS=rc-x
```

**Pull request merged**